### ISSUE TITLE
Improve keyboard navigation

### ISSUE THREAD

**Issue by lorant-one** [once-ui-system/nextjs-starter#6]
Some components can only be navigated with tabs, some get tabindex unintentionally.
Some components doesn't work properly with the return key.

**Affected components**

Components using DropdownWrapper: multiple tabindex on the main element.
SegmentedControl: Missing arrow key navigation
UserMenu: Return doesn't trigget dropdown (focus ring shape is incorrect on element)

All components should be reviewed to ensure proper accessibility.

**Comment by lightyfr** [comment]
Also, Id like to see the Dialog close when clicked outside of the form if that's possible.

**Comment by lorant-one** [comment]
@lightyfr just implemented this yesterday - gonna be included in 0.5 :)

**Comment by lorant-one** [comment]
fixed in 0.5 release
